pkg/lightning/backend/kv/context.go

```diff
@@ -0,0 +1,260 @@
+		userVars:          userVars,
+	}, nil
+}
+
+// SetUserVarVal sets the value of a user variable.
+func (ctx *litExprContext) SetUserVarVal(name string, dt types.Datum) {
+	ctx.userVars.SetUserVarVal(name, dt)
+}
+
+// UnsetUserVar unsets a user variable.
+func (ctx *litExprContext) UnsetUserVar(varName string) {
+	ctx.userVars.UnsetUserVar(varName)
+}
+
+// litTableMutateContext implements the `table.MutateContext` interface for lightning import.
+type litTableMutateContext struct {
+	exprCtx               *litExprContext
+	encodingConfig        tbctx.RowEncodingConfig
+	mutateBuffers         *tbctx.MutateBuffers
+	shardID               *variable.RowIDShardGenerator
+	reservedRowIDAlloc    stmtctx.ReservedRowIDAlloc
+	enableMutationChecker bool
+	assertionLevel        variable.AssertionLevel
+	tableDelta            struct {
+		sync.Mutex
+		m map[int64]variable.TableDelta
+	}
+}
+
+// AlternativeAllocators implements the `table.MutateContext` interface.
+func (*litTableMutateContext) AlternativeAllocators(*model.TableInfo) (autoid.Allocators, bool) {
+	// lightning does not support temporary tables, so we don't need to provide alternative allocators.
+	return autoid.Allocators{}, false
+}
+
+// GetExprCtx implements the `table.MutateContext` interface.
+func (ctx *litTableMutateContext) GetExprCtx() exprctx.ExprContext {
+	return ctx.exprCtx
+}
+
+// ConnectionID implements the `table.MutateContext` interface.
+func (*litTableMutateContext) ConnectionID() uint64 {
+	// Just return 0 because lightning import does not in any connection.
+	return 0
+}
+
+// InRestrictedSQL implements the `table.MutateContext` interface.
+func (*litTableMutateContext) InRestrictedSQL() bool {
+	// Just return false because lightning import does not in any SQL.
+	return false
+}
+
+// TxnAssertionLevel implements the `table.MutateContext` interface.
+func (ctx *litTableMutateContext) TxnAssertionLevel() variable.AssertionLevel {
+	return ctx.assertionLevel
+}
+
+// EnableMutationChecker implements the `table.MutateContext` interface.
+func (ctx *litTableMutateContext) EnableMutationChecker() bool {
+	return ctx.enableMutationChecker
+}
+
+// GetRowEncodingConfig implements the `table.MutateContext` interface.
+func (ctx *litTableMutateContext) GetRowEncodingConfig() tbctx.RowEncodingConfig {
+	return ctx.encodingConfig
+}
+
+// GetMutateBuffers implements the `table.MutateContext` interface.
+func (ctx *litTableMutateContext) GetMutateBuffers() *tbctx.MutateBuffers {
+	return ctx.mutateBuffers
+}
+
+// GetRowIDShardGenerator implements the `table.MutateContext` interface.
+func (ctx *litTableMutateContext) GetRowIDShardGenerator() *variable.RowIDShardGenerator {
+	return ctx.shardID
+}
+
+// GetReservedRowIDAlloc implements the `table.MutateContext` interface.
+func (ctx *litTableMutateContext) GetReservedRowIDAlloc() (*stmtctx.ReservedRowIDAlloc, bool) {
+	return &ctx.reservedRowIDAlloc, true
+}
+
+// GetBinlogSupport implements the `table.MutateContext` interface.
```

Comment:
this method will be soon removed by https://github.com/pingcap/tidb/pull/55955

Reply:
Yes, if #55955 merged first, I'll remove this implementation. Otherwise, #55955 should remove this ...